### PR TITLE
Added Helm chart to install plugin

### PR DESCRIPTION
**Pull request by sabre1041**
Added Helm chart to install plugin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,29 +128,20 @@ to run in-cluster.
...
-Once deployed, patch the
-[Console operator](https://github.com/openshift/console-operator)
-config to enable the plugin.
+Install the chart using the name of the plugin as the Helm release name into a new namespace or an existing namespace as specified by the `my-plugin-namespace` parameter by using the following command:
 
-```sh
-oc patch consoles.operator.openshift.io cluster \
-  --patch '{ "spec": { "plugins": ["my-plugin"] } }' --type=merge
+```shell
+helm upgrade -i  my-plugin charts/openshift-console-plugin -n my-plugin-namespace --create-namespace
```
Comment (spadgett) [inline README.md:144]: For this command to succeed, we need to supply an image here, correct? Is this typically done by editing values.yaml?

We probably want to make that a little more clear and give a recommendation (whether it's editing values.yaml or using `--set`).

Comment (spadgett) [reply README.md:144]: I did receive these warnings trying this due to the new pod security admission controller in 4.12.

```
W0825 08:02:06.016212   45304 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "console-plugin-template" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "console-plugin-template" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "console-plugin-template" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "console-plugin-template" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
W0825 08:02:06.330977   45304 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "console-plugin-template-patcher" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "console-plugin-template-patcher" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "console-plugin-template-patcher" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "console-plugin-template-patcher" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

**Review by spadgett — COMMENTED** [review]
Thanks @sabre1041 ! This is a clear improvement 👍 

@opayne1 FYI, when we merge this, we'll need to update the instructions in the OpenShift docs here:

https://docs.openshift.com/container-platform/4.11/web_console/dynamic-plug-ins.html#deploy-on-cluster_dynamic-plugins

File: charts/openshift-console-plugin/templates/patch-consoles-job.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          securityContext:
+            {{ tpl (toYaml .Values.plugin.jobs.patchConsoles.containerSecurityContext | indent 8) . }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.plugin.jobs.patchConsoles.resources | nindent 12 }}
+          command:
+            - /bin/bash
+            - -c
+            - |
+                oc patch consoles.operator.openshift.io cluster --patch '{ "spec": { "plugins": ["{{ template "openshift-console-plugin.name" . }}"] } }' --type=merge
```
Comment (spadgett) [inline charts/openshift-console-plugin/templates/patch-consoles-job.yaml:40]: I believe this will overwrite other enabled plugins :/ We should check.

**Comment by sabre1041** [comment]
@spadgett Each of the concerns mentioned previously have been address. Please review when you have a moment

File: charts/openshift-console-plugin/templates/patch-consoles-job.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      serviceAccountName: {{ template "openshift-console-plugin.patcherServiceAccountName" . }}
+      {{- if  and (.Values.plugin.securityContext.enabled) (.Values.plugin.jobs.patchConsoles.podSecurityContext.enabled) }}
+      securityContext: {{ tpl (toYaml (omit .Values.plugin.jobs.patchConsoles.podSecurityContext "enabled")) $ | nindent 8 }}
+      {{- end }}
+      terminationGracePeriodSeconds: 400
+      dnsPolicy: ClusterFirst
+      containers:
+        - name: {{ template "openshift-console-plugin.patcherName" . }}
+          image: {{ required "Patcher image must be specified!" .Values.plugin.jobs.patchConsoles.image }}
+          {{- if  and (.Values.plugin.disableSecurityContext) (.Values.plugin.jobs.patchConsoles.containerSecurityContext.enabled) }}
```
Comment (spadgett) [inline charts/openshift-console-plugin/templates/patch-consoles-job.yaml:29]:
```suggestion
          {{- if  and (.Values.plugin.securityContext.enabled) (.Values.plugin.jobs.patchConsoles.containerSecurityContext.enabled) }}
```

**Comment by sabre1041** [comment]
@spadgett corrected the above issue. thanks for catching it!

**Pull request merged**